lib/storage/search_options.go

```diff
@@ -0,0 +1,40 @@
+package storage
+
+import (
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/VictoriaMetrics/metrics"
+)
+
+// searchOptions contains common options for Search.
+type searchOptions struct {
+	deadline uint64
+
+	readMetricIDs *atomic.Uint64
```

Comment:
AFAIK atomic.Uint64 could be used as a value (without ref)

Reply:
Thanks for noticing, updated in https://github.com/VictoriaMetrics/VictoriaMetrics/pull/7514/commits/9fa43d55254f67b6b4692f5d7c059627635f0a54